src/test/java/com/getindata/connectors/http/internal/table/lookup/querycreators/GenericJsonAndUrlQueryCreatorTest.java

```diff
@@ -0,0 +1,235 @@
+                                getTableContext(config,
+                                        resolvedSchema)
+                        );
+        var row = new GenericRowData(2);
+        row.setField(0, StringData.fromString(value));
+        row.setField(1, StringData.fromString(value));
+        var createdQuery = genericJsonAndUrlQueryCreator.createLookupQuery(row);
+        // THEN
+        assertThat(createdQuery.getPathBasedUrlParameters().get("AAA")).isEqualTo(value);
+        assertThat(createdQuery.getBodyBasedUrlQueryParameters()).isEmpty();
+        assertThat(createdQuery.getLookupQuery()).isEqualTo(key1 + "=" + value
+                   + "&" + key2 + "=" + value);
+    }
+    @Test
+    public void failserializationOpenTest() {
+        List<String> paths_config =List.of("key1");
+        final String operation = "GET";
+        final String key = "key1";
+
+        LookupRow lookupRow = new LookupRow()
+                .addLookupEntry(
+                        new RowDataSingleValueLookupSchemaEntry(
+                                key,
+                                RowData.createFieldGetter(DataTypes.STRING().getLogicalType(), 0)
+                        ));
+        DataType dataType = row(List.of(
+                DataTypes.FIELD(key, DataTypes.STRING())
+        ));
+        ResolvedSchema resolvedSchema = ResolvedSchema.of(Column.physical(key,
+                DataTypes.STRING()));
+        Configuration config = new Configuration();
+        config.set(GenericJsonAndUrlQueryCreatorFactory.REQUEST_QUERY_PARAM_FIELDS, paths_config);
+        config.setString(LOOKUP_METHOD, operation);
+
+        lookupRow.setLookupPhysicalRowDataType(dataType);
+
+        GenericJsonAndUrlQueryCreator genericJsonAndUrlQueryCreator =
+                (GenericJsonAndUrlQueryCreator) new GenericJsonAndUrlQueryCreatorFactory()
+                        .createLookupQueryCreator(
+                                config,
+                                lookupRow,
+                                getTableContext(config,
+                                        resolvedSchema)
+                        );
+        // create a SerializationSchema that throws and exception in open
+        SerializationSchema<RowData> mockSerialiationSchema = new SerializationSchema<RowData>() {
+            @Override
+            public void open(InitializationContext context) throws Exception {
+                throw new Exception("Exception for testing");
+            }
+            @Override
+            public byte[] serialize(RowData element) {
+                return new byte[0];
+            }
+        };
+        genericJsonAndUrlQueryCreator.setSerializationSchema(mockSerialiationSchema);
+        var row = new GenericRowData(1);
+        assertThrows(RuntimeException.class, () -> {
+            genericJsonAndUrlQueryCreator.createLookupQuery(row);
+        });
+    }
+    @Test void convertToQueryParametersUnsupportedEncodingTest() {
+        // GIVEN
```

Comment:
nit: I would be nice if all tests have GIVEN WHEN THEN comments added.

Reply:
I added more where I thought it was appropriate. Let me know if there are any particular ones you are looking for that I have not done.